### PR TITLE
DOC: Add details on example contribution

### PR DESCRIPTION
**Pull request by eitanlees**
I added some instructions on contributing examples to the project. 

Addresses part of https://github.com/altair-viz/altair/issues/1554

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -11,4 +11,116 @@ visualization.
...
+Then create a branch for your example submission
+```
+git checkout -b <branch-name>
+```
+Now you are on a new branch where you can add your example file. All examples
+files are kept in the `altair/examples/` directory. Make sure the style and
+format match the other examples.  An official style guide for examples has not
+been created yet, but a few suggestions are 
+- Underline the title in the docstring with either `---` or `===`. Try to match
+  the number of characters.
```
Comment (jakevdp) [inline CONTRIBUTING.md]: More specifically, something like "The file docstring  will be rendered into HTML via [reStructuredText](http://docutils.sourceforge.net/rst.html), so use that format for any hyperlinks or text styling. In particular, be sure you include a title in the docstring underlined with `---`, and be sure that the size of the underline exactly matches the size of the title text."

**Comment by jakevdp** [comment]
This is a really great idea - thanks for putting it together! A couple minor comments, then we can merge.

**Comment by eitanlees** [comment]
I am not sure why this is failing the Travis build. I don't like that red x! 
 I am new to the world of continuous integration tools.

**Comment by jakevdp** [comment]
Test failure looks like a timeout in Travis, unrelated to this PR. I'll go ahead and merge – thanks!

**Pull request merged**